pkg/kubelet/server/streaming/errors.go

```diff
@@ -33,12 +38,29 @@ func ErrorTimeout(op string, timeout time.Duration) error {
 	return grpc.Errorf(codes.DeadlineExceeded, fmt.Sprintf("%s timed out after %s", op, timeout.String()))
 }
 
-// Translates a CRI streaming error into an HTTP status code.
-func HTTPStatus(err error) int {
+// The error returned when the maximum number of in-flight requests is exceeded.
+func ErrorTooManyInFlight(retryAfter int) error {
+	return grpc.Errorf(codes.ResourceExhausted, "maximum number of in-flight requests exceeded [Retry-After:%d]",
+		retryAfter)
+}
+
+// Translates a CRI streaming error into an appropriate HTTP response.
+func WriteError(err error, w http.ResponseWriter) error {
+	var status int
 	switch grpc.Code(err) {
 	case codes.NotFound:
-		return http.StatusNotFound
+		status = http.StatusNotFound
+	case codes.ResourceExhausted:
+		// Extract the Retry-After time from the error description.
+		retryAfter := retryAfterRegexp.FindStringSubmatch(err.Error())
```

Comment:
Thinking about this more, I think we're way over-engineering what should be a pretty exceptional case. I changed it to simply wait the TTL, and added a comment to this effect.